drivers/gpio/gpio_stm32.c

```diff
@@ -221,6 +221,49 @@ int gpio_stm32_configure(uint32_t *base_addr, int pin, int conf, int altf)
 	return 0;
 }
 
+/**
+ * @brief GPIO port clock handling
+ */
+int gpio_stm32_clock_request(const struct device *dev, bool on)
+{
+	const struct gpio_stm32_config *cfg = dev->config;
+	int ret = 0;
+
+	if (dev == NULL) {
+		ret = -ENODEV;
+		goto exit;
```

Comment:
`goto` can be avoided in this function, I'd use only in case resources need to be freed

Reply:
Do you mean we should exit directly ?

Reply:
yes